designer/client/components/DataPrettyPrint/DataPrettyPrint.tsx

```diff
@@ -15,23 +16,23 @@ export function componentToString(component) {
 }
 
 export function DataPrettyPrint(props) {
-  const { data } = props;
+  const { data } = useContext(DataContext);
   const { sections = [], pages = [] } = data;
 
   const model = {};
 
   pages.forEach((page) => {
-    page.components.forEach((component) => {
+    page.components!.forEach((component) => {
```

Comment:
is it worth doing a null check or are we certain components wont ever be null?

Reply:
Same for the others

Reply:
Might just be the typescript compilier